iphone/cli/lib/process-js-task.js

```diff
@@ -0,0 +1,374 @@
+			filename: from,
+			minify,
+			transpile,
+		});
+
+		try {
+			const modified = jsanalyze.analyzeJs(source, analyzeOptions);
+			const newContents = modified.contents;
+
+			// we want to sort by the "to" filename so that we correctly handle file overwriting
+			this.data.tiSymbols[to] = modified.symbols;
+
+			const dir = path.dirname(to);
+			fs.ensureDirSync(dir);
+
+			this.logger.debug(__('Copying and minifying %s => %s', from.cyan, to.cyan));
+			fs.writeFileSync(to, newContents);
+			this.builder.jsFilesChanged = true;
+
+			done();
+		} catch (err) {
+			err.message.split('\n').forEach(m => this.logger.error(m));
+			this.logger.error(err.stack);
+			if (err.codeFrame) { // if we have a nicely formatted pointer to syntax error from babel, use it!
+				this.logger.error(err.codeFrame);
+			}
+			process.exit(1);
+		} finally {
+			this.builder.unmarkBuildDirFile(to);
+		}
+	}
+
+	/**
+	 * Filters the given map of changed files for specific states and returns a
+	 * list of matchings paths.
+	 *
+	 * @param {Map<String, String>} changedFiles Map of changed file paths and their state
+	 * @param {String|Array} states Single state string or array of states to filter
+	 * @return {Array<String>}
+	 */
+	filterFilesByStatus(changedFiles, states) {
+		states = Array.isArray(states) ? states : [ states ];
+		const filteredFiles = [];
+		changedFiles.forEach((fileState, filePath) => {
+			if (states.includes(fileState)) {
+				filteredFiles.push(filePath);
+			}
+		});
+		return filteredFiles;
+	}
+
+	/**
+	 * Resolves th given full path to a relative path under it will be available in the app.
+	 *
+	 * @param {String} fullPath Full path used to resolve the relative path.
+	 * @param {Object} jsFiles Optional lookup object. Defaults to this.jsFiles.
+	 * @return {String}
+	 */
+	resolveRelativePath(fullPath, jsFiles) {
+		jsFiles = jsFiles || this.jsFiles;
+		return Object.keys(jsFiles).find(relativePath => {
+			return jsFiles[relativePath].src === fullPath;
+		});
+	}
+
+	/**
+	 * Handle deleted files by cleaning up the internal task data
+	 *
+	 * This will:
+	 *
+	 * - Remove file from the list of bootstrap files (if it's a .bootstrap.js file)
+	 * - Delete it's content hash
+	 * - Delete used Ti symbols
+	 *
+	 * @param {String} filePathAndName Full path to the deleted file
+	 * @return {Promise}
+	 */
+	handleDeletedFile(filePathAndName) {
+		let file = this.resolveRelativePath(filePathAndName, this.data.jsFiles);
+		const bootstrapPath = file.substr(0, file.length - 3);  // Remove the ".js" extension.
+		if (bootstrapPath.endsWith('.bootstrap')) {
+			const index = this.jsBootstrapFiles.indexOf(bootstrapPath);
+			if (index !== -1) {
+				this.jsBootstrapFiles.splice(index, 1);
+			}
+		}
+
+		const info = this.data.jsFiles[file];
+		if (info) {
+			delete this.data.contentHashes[info.src];
+			delete this.data.tiSymbols[info.dest];
+		}
+
+		return Promise.resolve();
```

Comment:
This method and return value seem self defeating. Doesn't this run sync and then return an empty Promise? Why not just make the method async so it basically wraps the body in a Promise? Or explicitly wrap the full body in a Promise?